### ISSUE TITLE
proposal: contrib/labstack/echo.v4: add NoDebugStack option

### ISSUE THREAD

**Issue by knusbaum** [DataDog/dd-trace-go#999]
It would be useful to add an `Option` to the echo integration similar to `NoDebugStack` from [goji](https://github.com/DataDog/dd-trace-go/blob/d62beaae492db2d8d56277e2fa97533e9f2e142e/contrib/zenazn/goji.v1/web/option.go#L50-L57) or [grpc](https://github.com/DataDog/dd-trace-go/blob/4cc670944564d33a6bca2cefa21226de0ef800f1/contrib/google.golang.org/grpc/option.go#L97-L103) and several others.

See: #987

**Comment by gbbr** [comment]
From the original issue:

> [...] And, I saw there is a trace stack for 4xx and 5xx status codes on the Datadog APM traces tag tab. [...]

Don't you think that this is the problem we need to fix? 4xx and 5xx responses should not return an error stack, or should they? Lastly, why this duplicate issue?

**Comment by knusbaum** [comment]
@gbbr 
I don't think it's possible to fix the original issue of stack traces not being complete. We've looked at that a number of times and it's something that we can't do without support from Go.

You're right, though, that 4xx status codes should not return an error stack. 5xx *should*. 

I opened this issue to avoid heavy editing of the other issue, or having the actual proposal be unclear or buried in the comments. I know often we convert the existing issue into a proposal, but that seemed messy in this case. I'm happy to close this and convert the other one if you feel strongly.
